arc-core/src/io/anuke/arc/files/FileHandle.java

```diff
@@ -688,8 +689,8 @@ public FileHandle child(String name){
                     }
 
                     @Override
-                    public FileHandle[] list(FilenameFilter filter){
-                        return Array.with(list()).select(f -> filter.accept(f.file, f.name())).toArray(FileHandle.class);
+                    public FileHandle[] list(FileFilter filter){
```

Comment:
This is the overload actually called by `FileDialog`.  Not sure if we should override all of the affected methods for completeness...